pandas/tests/extension/test_numpy.py

```diff
@@ -350,9 +350,9 @@ def test_fillna_fill_other(self, data_missing):
 
 
 class TestReshaping(BaseNumPyTests, base.BaseReshapingTests):
-    @skip_nested
+    @pytest.mark.skip(reason="Incorrect expected.")
```

Comment:
why did this change?

Reply:
I don't fully understand how testing with the PandasArrays works, but, within the `merge` implementation, we set a column in the resulting dataframe (the column with the key values) using an Index. 
And doing an `extract_array` on an Int64Index gives a PandasDtype("int64") column when running those tests. And because in this PR I changed it such that the values being set are not converted to a 1D array before ending up in BlockManager.iset, the extension dtype is now preserved. And so the expected result from the base test is wrong

Reply:
this turns out to be a PITA in a bunch of places.  could just monkeypatch extract_array to extract PandasArray too